samples/Common.c

```diff
@@ -864,9 +864,12 @@ STATUS getIceCandidatePairStatsCallback(UINT32 timerId, UINT64 currentTime, UINT
 
     pSampleConfiguration->rtcIceCandidatePairMetrics.requestedTypeOfStats = RTC_STATS_TYPE_CANDIDATE_PAIR;
 
-    // We need to execute this under the object lock due to race conditions that it could pose
-    MUTEX_LOCK(pSampleConfiguration->sampleConfigurationObjLock);
-    locked = TRUE;
+    // Use MUTEX_TRYLOCK to avoid possible dead lock when canceling timerQueue
+    if (!MUTEX_TRYLOCK(pSampleConfiguration->sampleConfigurationObjLock)) {
+        return retStatus;
```

Comment:
Per code conventions, I would recommend using the following construct
CHK(FALSE, retStatus);

Reply:
thought so but preferred to copy & paste the code from pregenerateCertTimerCallback()
should i change both?

Reply:
I would recommend keeping the changes as local as possible. I don't see this as a major issue and for some reason there is less consistency in the samples than in the library code.

Reply:
ok, then lets keep the copy & pasted code